### PR TITLE
WIP: Change the way cypress tests run

### PR DESCRIPTION
**Pull request by edewit**
now creates a test realm before starting the tests and logs in only for the test case instead of each time

### CODE REVIEW

File: cypress/support/util/keycloak_hooks.ts [hunk 1]
```diff
@@ -1,15 +1,25 @@
+import AdminClient from "./AdminClient";
+
+export const TEST_REALM = "test" as const;
```
Comment (jonkoops) [inline cypress/support/util/keycloak_hooks.ts:3]: Constant variables that are assigned a string literal are already a constant type by default.
```suggestion
export const TEST_REALM = "test";
```

File: cypress/support/util/AdminClient.ts [hunk 2]
```diff
@@ -22,6 +24,12 @@ export default class AdminClient {
 
   async createRealm(realm: string) {
     await this.login();
+    const theRealm = (await this.client.realms.findOne({ realm })) as
```
Comment (jonkoops) [inline cypress/support/util/AdminClient.ts:27]: Looks like the return type of the `findOne()` method is incorrect, can we fix this upstream in the Admin Client?

File: cypress/support/pages/admin_console/manage/realm_settings/RealmSettingsPage.ts [hunk 3]
```diff
@@ -226,7 +226,7 @@ export default class RealmSettingsPage {
   }
 
   toggleAddProviderDropdown() {
-    const keysUrl = "/auth/admin/realms/master/keys";
+    const keysUrl = "/auth/admin/realms/test/keys";
```
Comment (jonkoops) [inline cypress/support/pages/admin_console/manage/realm_settings/RealmSettingsPage.ts:229]: Perhaps we can use the `TEST_REALM` constant here as well?

File: cypress/support/pages/admin_console/manage/realm_roles/AssociatedRolesPage.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ export default class AssociatedRolesPage {
   addAssociatedRealmRole() {
     cy.findByTestId(this.actionDropdown).last().click();
 
-    const load = "/auth/admin/realms/master/clients";
+    const load = "/auth/admin/realms/test/clients";
```
Comment (jonkoops) [inline cypress/support/pages/admin_console/manage/realm_roles/AssociatedRolesPage.ts:15]: Perhaps we can use the `TEST_REALM` constant here as well?

File: cypress/support/pages/admin_console/manage/providers/ProviderPage.ts [hunk 5]
```diff
@@ -203,13 +203,9 @@ export default class ProviderPage {
 
   createRole(roleName: string) {
     cy.get(this.rolesTab).click();
-    cy.wait(1000);
```
Comment (jonkoops) [inline cypress/support/pages/admin_console/manage/providers/ProviderPage.ts:206]: Nice cleanup!

File: cypress/support/pages/admin_console/manage/events/AdminEventsTab.ts [hunk 6]
```diff
@@ -45,7 +45,7 @@ export default class AdminEventsTab {
     cy.findByTestId(this.searchAdminEventDrpDwnBtn).click();
     cy.findByTestId(this.resourcePathInputFld).type("events/config");
 
-    cy.intercept("/auth/admin/realms/master/admin-events*").as("eventsFetch");
+    cy.intercept("/auth/admin/realms/test/admin-events*").as("eventsFetch");
```
Comment (jonkoops) [inline cypress/support/pages/admin_console/manage/events/AdminEventsTab.ts:48]: Perhaps we can use the `TEST_REALM` constant here as well?

File: cypress/support/pages/admin_console/manage/identity_providers/AddMapperPage.ts [hunk 7]
```diff
@@ -70,7 +70,7 @@ export default class AddMapperPage {
   }
 
   addRoleToMapperForm() {
-    const load = "/auth/admin/realms/master/roles";
+    const load = "/auth/admin/realms/test/roles";
```
Comment (jonkoops) [inline cypress/support/pages/admin_console/manage/identity_providers/AddMapperPage.ts:73]: Perhaps we can use the `TEST_REALM` constant here as well?

File: cypress/support/pages/admin_console/ListingPage.ts [hunk 8]
```diff
@@ -25,7 +25,7 @@ export default class ListingPage {
 
   searchItem(searchValue: string, wait = true) {
     if (wait) {
-      const searchUrl = `/auth/admin/realms/master/*${searchValue}*`;
+      const searchUrl = `/auth/admin/realms/test/*${searchValue}*`;
```
Comment (jonkoops) [inline cypress/support/pages/admin_console/ListingPage.ts:28]: Perhaps we can use the `TEST_REALM` constant here as well?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: cypress/support/util/AdminClient.ts (hunk 2 above)
Comment (edewit) [reply cypress/support/util/AdminClient.ts:27]: for sure

**Pull request closed without merging**